server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -203,13 +205,22 @@ private void executeIngestAndBulk(Task task, final BulkRequest bulkRequest, fina
         final ActionListener<BulkResponse> listener, final AtomicArray<BulkItemResponse> responses,
         Map<String, IndexNotFoundException> indicesThatCannotBeCreated) {
         boolean hasIndexRequestsWithPipelines = false;
-        ImmutableOpenMap<String, IndexMetaData> indicesMetaData = clusterService.state().getMetaData().indices();
+        final ClusterState state = clusterService.state();
```

Comment:
NIT: You can shorten the logic a little by caching the metadata `clusterService.state().getMetaData()` instead of getting it from the state twice.

Reply:
done.